### ISSUE TITLE
Questions about Mask h (x)

### ISSUE THREAD

**Issue by zhaoxueyi6** [AslanDing/AutoTCL#3]
Hello author，I have a question that I seem to have not found the position of the mask h in get_features to decompose the time series x into the informative part x *. Could you please let me know.I see that it seems that x has been directly enhanced to form ax.
    def get_features(self, x, training = True, n_epochs=-1, mask=None):  

        embedding = self.augsharenet(x)   
        weight_h = self.factor_augnet(embedding)  
        weight_s = self.augmentation_projector(embedding)  

        
        mask_h = self._sample_graph(weight_h,training= training)  #输出（32，257，8）

        if self.hard_mask:
            hard_mask_h = (torch.sign(mask_h-0.5)+1)/2
            # print(hard_mask_h)
#             mask_h = (mask_h-hard_mask_h).detach()+hard_mask_h
            mask_h = (hard_mask_h - mask_h).detach()+mask_h
        ax = weight_s * mask_h * x  # augmented x'

        
        if torch.isnan(ax).any() or torch.isnan(x).any():
            exit(1)

        # note: I add mask
        out1 = self._net(x,mask)  # representation
        out2 = self._net(ax,mask)  # representation of augmented x'
        return x, ax, out1, out2, weight_h

**Comment by AslanDing** [comment]
The informative part is mask_h*x.   It is part of  "  ax = weight_s * mask_h * x  "  Thanks.

**Comment by zhaoxueyi6** [comment]
I understand, thank you!！

**Comment by zhaoxueyi6** [comment]
I understand, thank you!！
&nbsp;


赵雪屹
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2025年2月20日(星期四) 中午11:59
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [AslanDing/AutoTCL] Questions about Mask h (x) (Issue #3)












   
The informative part is mask_h*x.   It is part of  "  ax = weight_s * mask_h * x  "  Thanks.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  AslanDing left a comment (AslanDing/AutoTCL#3)
 
The informative part is mask_h*x.   It is part of  "  ax = weight_s * mask_h * x  "  Thanks.
